maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1624,20 +1643,65 @@ static void populateProperties(
-            String name = (String) e.getKey();
-            String value = interpolator.interpolate((String) e.getValue());
-            userProperties.setProperty(name, value);
-            // ----------------------------------------------------------------------
-            // I'm leaving the setting of system properties here as not to break
-            // the SystemPropertyProfileActivator. This won't harm embedding. jvz.
-            // ----------------------------------------------------------------------
-            if (System.getProperty(name) == null) {
-                System.setProperty(name, value);
-            }
+        // ----------------------------------------------------------------------
+        // Options that are set on the command line become system properties
+        // and therefore are set in the session properties. System properties
+        // are most dominant.
+        // ----------------------------------------------------------------------
+
+        Properties userSpecifiedProperties =
+                commandLine.getOptionProperties(String.valueOf(CLIManager.SET_USER_PROPERTY));
+        userProperties.putAll(userSpecifiedProperties);
+
+        // ----------------------------------------------------------------------
+        // Load config files
+        // ----------------------------------------------------------------------
+        Function<String, String> callback =
+                or(paths::getProperty, prefix("cli.", commandLine::getOptionValue), systemProperties::getProperty);
+
+        Path mavenConf;
+        if (systemProperties.getProperty(MAVEN_INSTALLATION_CONF) != null) {
+            mavenConf = fileSystem.getPath(systemProperties.getProperty(MAVEN_INSTALLATION_CONF));
+        } else if (systemProperties.getProperty("maven.conf") != null) {
+            mavenConf = fileSystem.getPath(systemProperties.getProperty("maven.conf"));
+        } else if (systemProperties.getProperty(MAVEN_HOME) != null) {
+            mavenConf = fileSystem.getPath(systemProperties.getProperty(MAVEN_HOME), "conf");
+        } else {
+            mavenConf = fileSystem.getPath("");
         }
+        Path propertiesFile = mavenConf.resolve("maven.properties");
+        MavenPropertiesLoader.loadProperties(userProperties, propertiesFile, callback, false);
+
+        // ----------------------------------------------------------------------
+        // I'm leaving the setting of system properties here as not to break
+        // the SystemPropertyProfileActivator. This won't harm embedding. jvz.
+        // ----------------------------------------------------------------------
+        Set<String> sys = SystemProperties.getSystemProperties().stringPropertyNames();
+        userProperties.stringPropertyNames().stream()
+                .filter(k -> !sys.contains(k))
+                .forEach(k -> System.setProperty(k, userProperties.getProperty(k)));
+    }
```

Comment:
We must seriously reconsider this promotion in the future before GA.

Reply:
The best we should not change a System property at all

Reply:
I've raised https://github.com/apache/maven/pull/1661 to assess if there are any failing ITs as a first step.

Reply:
Invoker and friends are our worst enemies here.